File: subi930/Programmers/L2/귤_고르기.java
// 1. 귤 배열 정렬
// 2. 중복 수 리스트 생성 및 저장
// 3. 중복수 리스트 정렬(오름차순)
// 4. k<중복수 리스트 요소 합 의 인덱스 + 1 = answer
import java.util.*;

class Solution {
    public int solution(int k, int [] tangerine) {
        int answer = 0;
        Integer[] tan = Arrays.stream(tangerine).boxed().toArray(Integer[]::new);
        Arrays.sort(tan, Collections.reverseOrder()); // 내림차순으로 정렬

        List<Integer> typeCount = new ArrayList<>();
        int count = 1;

        for (int i = 1; i < tan.length; i++) {
            if (tan[i - 1]==tan[i]) {
                count++;
            }else {
                typeCount.add(count);
                count = 1;
            }
        }
        //typeCount.add(count);  // 마지막 count 추가

        int sum = 0, idx = 0;
        while (sum < k && idx < typeCount.size()) {
            sum += typeCount.get(idx);
            idx++;
        }

        answer = idx-1;
        return answer;
    }
}

File: xeunnie/Programmers/L2/귤 고르기.js
function solution(k, tangerine) {
    const countMap = new Map();

    for (let size of tangerine) {
        countMap.has(size) ?
        countMap.set(size, countMap.get(size) + 1):
        countMap.set(size, 1);
    }

    const counts = Array.from(countMap.values());
    
    counts.sort((a, b) => b - a);
    
    let totalKinds = 0;
    let totalTangerines = 0;
    
    for (let count of counts) {
        totalTangerines += count;
        totalKinds++;
        if (totalTangerines >= k) {
            break;
        }
    }
    
    return totalKinds;
}
